File: Javascript/aoc_2015_d1_b.js
/* 
* Read the input file, store into a string and split
* it to a character array. For each character set the 
* floor value accordingly. The first time it becomes 
* negative set the position value and set the flag so 
* it is not calculated again. When reach the end of the
* array, break the loop and log the final floor and
* position value. 
*/
var result = function(){
    const fs = require('fs');
    var file = "../_Input/aoc_2015_d1_input.txt";
    var str = fs.readFileSync(file).toString();
    var chars = str.split('');
    var c;
    var floor = 0;
    var position = 0;
    var enteredBasementFlag = false;
    for (c in chars)
    {
        if (chars[c] === '(')
            floor += 1
        else if (chars[c] === ')')
            floor += -1

        if (floor === -1 && enteredBasementFlag === false)
        {
            position = parseInt(c) + 1; // zero based index
            enteredBasementFlag = true;
        }

    }
    console.log("The final floor is : " + floor );
    console.log("The first time entered the first basement is : " + position );
}

result();

File: Python/aoc_2015_d1_b.py
# Read each character of the input file, set the 
# floor value accordingly and check if the floor
# value is negative. The first time it becomes negative
# print the position value and set the flag so it is not
# printed again. When reach the end of the
# file, break the loop and print the final floor value.

floor = 0
position = 0
enteredBasementFlag = False

with open("input.txt") as inputFile:
    while True:

        character = inputFile.read(1)
        position += 1

        if not character:
            print("Final Floor is : ", floor)
            break
        elif character == '(' :
            floor += 1
        elif character == ')' :
            floor -= 1

        if floor < 0 and not enteredBasementFlag :
            enteredBasementFlag = True
            print("Position first entered first basement is : ", position)   